.github/hooks/clang-format-check.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+RANGE=""
+
+while [[ $# -gt 0 ]]; do
+  echo $1
+  echo $2
+  case "$1" in
+  --range)
+    RANGE="$2"
+    shift 2
+    ;;
+  *)
+    echo "Unknown arg $1" >&2
+    exit 64
+    ;;
+  esac
+done
+
+regex='\.(c|cc|cpp|cxx|h|hh|hpp|hxx)$'
```

Comment:
The changes to affected files broke at least https://github.com/ROCm/rocm-systems/blob/develop/projects/clr/rocclr/device/pal/palblitcl.cpp (this is already being fixed).

If using pre-commit instead of a custom script, individual file or extension exclusions would be easier to author.